### ISSUE TITLE
Use relative `import`

### ISSUE THREAD

**Issue by LecrisUT** [teemtee/tmt#3118]
I just realized the cause of the failure of https://github.com/teemtee/tmt/actions/runs/9726385087/job/26844764885?pr=3062

Basically because the project does not use `src-layout` and absolute `import` paths are used, it is very easy to be out-of-sync, e.g. in that PR the dependencies and entry_point come from the `pre-commit`, but the source for `tmt` module comes from the current directory instead. #3117 intends to fix part of that issue, but if other users have a `tmt` module in their top-level, than things will get wacky again.

**Comment by psss** [comment]
@happz agreed to provide some interesting articles because the relative import also has some disadvantages.

**Comment by LecrisUT** [comment]
Let's try to discuss this one soon. I do every now and then have issues which I suspect that are coming from the non-src-layout format.

**Comment by therazix** [comment]
From the hacking session: Let's start with the relative imports first.
